### PR TITLE
solution

### PR DESCRIPTION
**Pull request by Vas35**

### CODE REVIEW

File: src/main/java/core/basesyntax/Car.java [hunk 1]
```diff
@@ -1,5 +1,28 @@
...
+        this.headlights = headlights;
+        this.tank = tank;
+        this.wheel = wheel;
+        this.engine = engine;
+    }
+
+    @Override
+    protected Car clone() throws CloneNotSupportedException {
+        Car car = new Car(glass, headlights, tank, wheel, engine.clone());
+        return car;
```
Comment (dmytro-hryhoruk) [inline src/main/java/core/basesyntax/Car.java:21]:
```suggestion
        return  new Car(glass, headlights, tank, wheel, engine.clone());
```
We don't need the local variable in this case

File: src/main/java/core/basesyntax/Engine.java [hunk 2]
```diff
@@ -1,5 +1,25 @@
...
+        this.volume = volume;
+        this.pressure = pressure;
+        this.speed = speed;
+        this.age = age;
+    }
+
+    @Override
+    protected Engine clone() throws CloneNotSupportedException {
+        Engine engine = new Engine(temperature, volume, pressure, speed, age);
+        return engine;
```
Comment (dmytro-hryhoruk) [inline src/main/java/core/basesyntax/Engine.java:21]: same

File: src/main/java/core/basesyntax/Engine.java [hunk 3]
```diff
@@ -1,5 +1,25 @@
...
+    public Engine(int temperature, int volume, int pressure, int speed, int age) {
+        this.temperature = temperature;
+        this.volume = volume;
+        this.pressure = pressure;
+        this.speed = speed;
+        this.age = age;
+    }
+
+    @Override
+    protected Engine clone() throws CloneNotSupportedException {
```
Comment (dmytro-hryhoruk) [inline src/main/java/core/basesyntax/Engine.java:19]: does it really throws that exception or it's has a grey color?

File: src/main/java/core/basesyntax/Car.java (hunk 1 above)
Comment (Vas35) [reply src/main/java/core/basesyntax/Car.java:21]: Ok, thanks I will do.

File: src/main/java/core/basesyntax/Engine.java (hunk 2 above)
Comment (Vas35) [reply src/main/java/core/basesyntax/Engine.java:21]: I understood.

File: src/main/java/core/basesyntax/Engine.java (hunk 3 above)
Comment (Vas35) [reply src/main/java/core/basesyntax/Engine.java:19]: What do you mean asking about grey color?

Comment (dmytro-hryhoruk) [reply src/main/java/core/basesyntax/Engine.java:19]: image

Comment (dmytro-hryhoruk) [reply src/main/java/core/basesyntax/Engine.java:19]: remove it if it's grey(which means it's never throws that exceptions) but probably you were using super.clone() and it has been left after that

File: src/main/java/core/basesyntax/Car.java [hunk 4]
```diff
@@ -1,5 +1,28 @@
...
+        this.glass = glass;
+        this.headlights = headlights;
+        this.tank = tank;
+        this.wheel = wheel;
+        this.engine = engine;
+    }
+
+    @Override
+    protected Car clone() {
+
```
Comment (dmytro-hryhoruk) [inline src/main/java/core/basesyntax/Car.java:20]:
```suggestion
```

File: src/main/java/core/basesyntax/Engine.java [hunk 5]
```diff
@@ -1,5 +1,25 @@
...
+        this.temperature = temperature;
+        this.volume = volume;
+        this.pressure = pressure;
+        this.speed = speed;
+        this.age = age;
+    }
+
+    @Override
+    protected Engine clone() {
+
```
Comment (dmytro-hryhoruk) [inline src/main/java/core/basesyntax/Engine.java:20]:
```suggestion
```

**Pull request closed without merging**